app/models/post.rb

```diff
@@ -0,0 +1,7 @@
+class Post < ApplicationRecord
+  has_many :comments
+  belongs_to :user
+  validates_associated :user
+  validates :body, presence: true, length: { maximum: 250 }
+  validates :username, presence: true, uniqueness: { case_sensitive: true }, length: { maximum: 50 }
```

Comment:
- Since the association between the user and post is established thru line 3, the post will not be created until a user_id is provided. Therefore line 4 is not necessary.
- You can access the username of the post when you query `post.user.username`. Therefore, you don't have to create a column of username in post table as proved on line 6